pkg/kubectl/cmd/create_pvc.go

```diff
@@ -0,0 +1,222 @@
+	cmd.Flags().String("capacity", "", "Capacity represents the minimum storage capacity required.")
+	cmd.Flags().String("storage-class", "", "Name of the StorageClass required by the claim.")
+	cmd.Flags().StringVar(&c.VolumeName, "volume-name", "", "VolumeName is the binding reference to the PersistentVolume backing this claim.")
+
+	return cmd
+}
+
+func (c *CreatePVCOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args []string) error {
+	name, err := NameFromCommandArgs(cmd, args)
+	if err != nil {
+		return err
+	}
+	c.Name = name
+
+	// Transform access mode abbreviation to complete access mode and remove duplicate access mode.
+	accessModes := []api.PersistentVolumeAccessMode{}
+	modes := cmdutil.GetFlagStringSlice(cmd, "access-mode")
+	for _, m := range modes {
+		var accessMode api.PersistentVolumeAccessMode
+		switch m {
+		case "RWX", string(api.ReadWriteMany):
+			accessMode = api.ReadWriteMany
+		case "RWO", string(api.ReadWriteOnce):
+			accessMode = api.ReadWriteOnce
+		case "ROX", string(api.ReadOnlyMany):
+			accessMode = api.ReadOnlyMany
+		default:
+			return fmt.Errorf("invalid access mode: %s", m)
+		}
+
+		containsMode := false
+		for _, a := range accessModes {
+			if a == accessMode {
+				containsMode = true
+				break
+			}
+		}
+
+		if !containsMode {
+			accessModes = append(accessModes, accessMode)
+		}
+	}
+	c.AccessModes = accessModes
+
+	// If user does not provide storage-class option, then StorageClass will be set as nil.
+	// Otherwise, StorageClass will be the value set by user, including empty string.
+	storageClassFlag := cmd.Flags().Lookup("storage-class")
+	if storageClassFlag != nil && storageClassFlag.Changed {
+		storageClass := cmdutil.GetFlagString(cmd, "storage-class")
+		c.StorageClass = &storageClass
+	}
+
+	capacityString := cmdutil.GetFlagString(cmd, "capacity")
+	if len(capacityString) != 0 {
+		quantity, err := resource.ParseQuantity(capacityString)
+		if err != nil {
+			return err
+		}
+
+		c.Capacity = &quantity
+	}
+
+	// Complete other options for Run.
+	c.Mapper, _ = f.Object()
+
+	c.DryRun = cmdutil.GetDryRunFlag(cmd)
+	c.OutputFormat = cmdutil.GetFlagString(cmd, "output")
+
+	c.Namespace, _, err = f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+
+	c.PrintObject = func(obj runtime.Object) error {
+		return f.PrintObject(cmd, false, c.Mapper, obj, c.Out)
+	}
+
+	clientSet, err := f.ClientSet()
+	if err != nil {
+		return err
+	}
+	c.Client = clientSet.Core().PersistentVolumeClaims(c.Namespace)
+
+	return nil
+}
+
+func (c *CreatePVCOptions) Validate() error {
+	if c.Name == "" {
+		return fmt.Errorf("name must be specified")
+	}
+
+	if len(c.AccessModes) == 0 {
+		return fmt.Errorf("at least one access mode must be specified")
+	}
+
+	if c.Capacity == nil {
+		return fmt.Errorf("capacity of the storage must be specified")
+	}
+
+	return nil
+}
+
+func (c *CreatePVCOptions) RunCreatePVC() error {
+	pvc := &api.PersistentVolumeClaim{}
+
```

Comment:
redundant newline?

Reply:
I'd like to use newlines to separate different blocks of code to make it more easier to read, but not sure if we need to follow some guidance on this, any references?